### ISSUE TITLE
ZC1090 false positive: quoted `=~` regex works in Zsh (Bash-only rule)

### ISSUE THREAD

**Issue by afadesigns** [afadesigns/zshellcheck#1368]
### Summary
ZC1090 warns on `[[ $x =~ "regex" ]]` with "Quoted regex pattern matches literally. Remove quotes to enable regex matching", and its description claims quoting "forces literal string matching in Zsh/Bash." This is Bash-only behavior. In Zsh the right-hand side of `=~` is always a regular expression regardless of quoting, so the warning is false and the "remove quotes" advice is harmful.

### Reproduction
```zsh
[[ "123" =~ "^[0-9]+$" ]] && echo MATCH   # MATCH (regex, not literal)
[[ "abc" =~ "^[0-9]+$" ]] && echo MATCH   # no match — confirms regex semantics
```
Bash, by contrast, treats the quoted RHS as a literal string.

### Why it is wrong
- `man zshmisc` documents no quote-based literal switch for `=~`; the rule exists only in `man bash`.
- Removing the quotes (the kata's advice) breaks any regex containing a space or `[[`-significant character: `[[ "a b" =~ ^a b$ ]]` is a `condition expected` syntax error.
- Quoting the `=~` RHS is idiomatic Zsh — zsh's own shipped completions (`_zypper`, `_transmission`) quote it to protect metacharacters from globbing and splitting.

### Expected
Stop flagging quoted `=~` patterns in Zsh (the premise is inverted and the advice harmful). Per the no-removal policy, neutralize the check and correct the description/messages and the mislabeled test fixtures.

### Environment
- zshellcheck v1.1.0
